cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -457,6 +457,10 @@ rebuildInstallPlan verbosity
                                                     compiler progdb platform
                                                     corePackageDbs
           sourcePkgDb       <- getSourcePackages    verbosity withRepoCtx
+                                                    (fmap fromIntegral
+                                                     . flagToMaybe
+                                                     . projectConfigIndexSnapshot
+                                                     $ projectConfigShared)
```

Comment:
The action here now depends on the value `projectConfigIndexSnapshot projectConfigShared` and so it must be part of the value that `rerunIfChanged` monitors. It's not appropriate to put into the `SolverSettings` type, since it actually affects the source package db that we run the solver with. So if it's not in the monitor value by inclusion in the `SolverSettings` type then it needs to be in the monitor value directly.

Stick it in a let/where and include it into the monitor value.